src/include/ipc4/dmic.h

```diff
@@ -0,0 +1,225 @@
+	 */
+	uint32_t out_gain_left;
+
+	/**
+	 * @brief OUT_GAIN_RIGHT
+	 * Output gain value for the right channel.
+	 */
+	uint32_t out_gain_rigth;
+
+	/**
+	 * @brief rsvd_2
+	 * Reserved field.
+	 */
+	uint32_t rsvd_2[2];
+} __packed __aligned(4);
+
+/**
+ * @brief Structure representing the configuration of the PDM control for DMIC.
+ *
+ * This structure defines the configuration parameters for the PDM control of the DMIC
+ * (Digital Microphone) module. It includes fields for controlling the CIC (Cascaded
+ * Integrator-Comb) filter, MIC (Microphone) control, SoundWire mapping, FIR (Finite
+ * Impulse Response) configurations, and FIR coefficients.
+ */
+struct dmic_pdm_ctrl_cfg {
+	/**
+	 * @brief CIC_CONTROL
+	 * Control register for CIC configuration.
+	 */
+	uint32_t cic_control;
+	/**
+	 * @brief CIC_CONFIG
+	 * Configuration register for CIC filter.
+	 */
+	uint32_t cic_config;
+	/**
+	 * @brief Reserved field
+	 */
+	uint32_t rsvd_0;
+	/**
+	 * @brief MIC_CONTROL
+	 * Control register for MIC configuration.
+	 */
+	uint32_t mic_control;
+	/**
+	 * @brief
+	 * This field is used on platforms with SoundWire, otherwise ignored.
+	 */
+	uint32_t pdmsm;
+	/**
+	 * @brief Index of another PDMCtrlCfg to be used as a source of FIR coefficients.
+	 */
+	uint32_t reuse_fir_from_pdm;
+	/**
+	 * @brief Reserved field
+	 */
+	uint32_t rsvd_1[2];
+	/**
+	 * @brief FIR configurations
+	 */
+	struct dmic_fir_cfg fir_config[2];
+	/**
+	 * @brief Array of FIR coefficients, channel A goes first, then channel B.
+	 */
+	uint32_t fir_coeffs[0];
+} __packed __aligned(4);
+
+/**
+ * @brief Structure representing the configuration blob for DMIC (Digital Microphone) settings.
+ *
+ * This structure contains various configuration settings for DMIC, including time-slot mappings,
+ * global configuration, PDM channel configuration, and PDM controller configuration.
+ */
+struct dmic_config_blob {
+	/**
+	 * @brief Time-slot mappings.
+	 */
+	uint32_t ts_group[4];
+
+	/**
+	 * @brief DMIC global configuration.
+	 */
+	struct dmic_global_cfg global_cfg;
+
+	/**
+	 * @brief PDM channels to be programmed using data from channel_cfg array.
+	 */
+	uint32_t channel_ctrl_mask : 8;
+
+	/**
+	 * @brief Clock source for DMIC.
+	 */
+	uint32_t clock_source : 8;
+
+	/**
+	 * @brief Reserved field.
+	 */
+	uint32_t rsvd : 16;
+
+	/**
+	 * @brief PDM channel configuration settings.
+	 */
+	struct dmic_channel_cfg channel_cfg[0];
+
+	/**
+	 * @brief PDM controllers to be programmed using data from pdm_ctrl_cfg array.
+	 */
+	uint32_t pdm_ctrl_mask;
+
+	/**
+	 * @brief PDM controller configuration settings.
+	 */
+	struct dmic_pdm_ctrl_cfg pdm_ctrl_cfg[0];
+} __packed __aligned(4);
+
+/**
+ * @brief Structure representing the configuration data for DMIC.
+ */
+struct dmic_config_data {
+	/**< Gateway attributes */
+	union ipc4_gateway_attributes gtw_attributes;
+	/**< DMIC Configuration BLOB */
+	struct dmic_config_blob dmic_blob;
+}  __packed __aligned(4);
```

Comment:
erm, we've been enabling DMIC w/ IPC4 and never needed this header.

Why do we need it now??

This looks like a duplication of existing information to me.